python/ray/serve/_private/deployment_state.py

```diff
@@ -1595,14 +1596,18 @@ def get_total_num_requests(self) -> int:
 
         total_requests = 0
         running_replicas = self._replicas.get([ReplicaState.RUNNING])
-        for replica in running_replicas:
-            replica_tag = replica.replica_tag
-            if replica_tag in self.replica_average_ongoing_requests:
-                total_requests += self.replica_average_ongoing_requests[replica_tag][1]
+        if RAY_SERVE_COLLECT_AUTOSCALING_METRICS_ON_HANDLE:
```

Comment:
```suggestion
        if RAY_SERVE_COLLECT_AUTOSCALING_METRICS_ON_HANDLE or len(running_replicas) == 0:
```

and drop the section below